app/Services/GitHub/GitHubApi.php

```diff
@@ -29,6 +33,56 @@ public function fetchPublicRepositories(string $username): Collection
         });
     }
 
+    public function latestVersionOnDate(string $repository, Carbon $onDate): string
+    {
+        [$organisation, $repository] = explode('/', $repository);
+
+        $api = $this->client->api('repo')->releases();
+
+        $paginator = new ResultPager($this->client);
+
+        $releases = $paginator->fetchAll($api, 'all', [$organisation, $repository]);
+
+        $latestAvailableReleaseOnDate = collect($releases)
+            ->first(function (array $releaseProperties) use ($onDate) {
+                $createdAt = Carbon::create($releaseProperties['created_at']);
+
+                return $createdAt->isBefore($onDate);
+            });
+
+        if (!$latestAvailableReleaseOnDate) {
+            throw new Exception("No release found for {$repository} on date {$onDate->format('Y-m-d')}");
+        }
+
+        $featureVersionNumber = Str::beforeLast($latestAvailableReleaseOnDate['tag_name'], '.');
+
+        $latestBugFixReleaseForFeatureVersionRelease = collect($releases)
+            ->first(function (array $releaseProperties) use ($featureVersionNumber) {
+                return str_starts_with($releaseProperties['tag_name'], $featureVersionNumber . '.');
+            });
+
+        return $latestBugFixReleaseForFeatureVersionRelease['tag_name'];
+    }
+
+    public function temporaryUrlOfLatestAvailableRelease(string $repository, Carbon $onDate = null)
+    {
+        $onDate = Carbon::createFromFormat('Y-m-d', '2020-10-03');
```

Comment:
This will always create a date for 2020-10-03, missing a null check?

Reply:
Whoops, that's a left debug statement.